### ISSUE TITLE
False positive in RightCurly with 'alone_or_singleline' RightCurlyOption in try/catch

### ISSUE THREAD

**Issue by guillerodriguez** [checkstyle/checkstyle#12020]
I have read check documentation: https://checkstyle.org/checks/blocks/rightcurly.html#RightCurly

Related documentation: https://checkstyle.org/property_types.html#RightCurlyOption

From `alone_or_singleline`:
> The brace must be alone on the line, yet single-line format of block is allowed. 

I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ javac RCurlyTest.java

/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
    "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
    "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="severity" value="error"/>
    <property name="tabWidth" value="4"/>
    <property name="fileExtensions" value="java, properties, xml"/>

    <module name="TreeWalker">
        <module name="RightCurly">
            <property name="option" value="alone_or_singleline"/>
        </module>
    </module>
</module>

/var/tmp $ cat RCurlyTest.java
public class RCurlyTest {
	public static void main(String args[]) {
		try { System.out.println(args[0]); } catch (Exception e) {}
	}
}

/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-10.3.1-all.jar -c config.xml RCurlyTest.java
Starting audit...
[ERROR] [...]/RCurlyTest.java:3:44: '}' at column 38 should be alone on a line. [RightCurly]
Audit done.
Checkstyle ends with 1 errors.
```
---

**Describe what you expect in detail.**

Since the whole try..catch construct is in a single line, I expect the RightCurly test to pass (i.e. no errors raised).

As a side-note. Column number in the error message seems to be wrong when tabs are used in the source file. In the above example you can see that:

 * RCurlyTest.java:3:44 is correct and honours the top level tabWidth property in the config.xml file
 * However the message incorrectly refers to "column 38"

**Comment by nick-mancuso** [comment]
@guillerodriguez thanks for your bug report, please follow the template and add a link to the affected check.

**Comment by guillerodriguez** [comment]
Sorry, I had overlooked that. Fixed now.

**Comment by guillerodriguez** [comment]
@nrmancuso Any updates? If this looks like a valid issue, perhaps you can approve it?

**Comment by nrmancuso** [comment]
@guillerodriguez thanks for pinging me, lost track of this one. I have added some additional details to your issue description and approved this issue. Thanks for your report.

**Comment by mahfouz72** [comment]
I am on it

**Comment by nrmancuso** [comment]
Closed via #14594
